### PR TITLE
Autodetect endpoint on BMP

### PR DESCRIPTION
**Pull request by robots**
Endpoint has been changed recently.
See https://github.com/blackmagic-debug/blackmagic/commit/6af8d5dc48d1ba0e8a0bb752bd40a4242d84765e

### CODE REVIEW

**Comment by zyp** [comment]
Since we apparently can't rely on it being a fixed number anymore, I'd be inclined to look it up in the configuration descriptor in a similar manner to what we do for Orbtrace Mini.

**Comment by robots** [comment]
Yes, this is what came to my mind right after i posted this PR. I will look into that.

File: Src/orbtraceIf.c
Comment (dragonmux) [inline Src/orbtraceIf.c:489]: Rather than assuming the interface number is always going to be the same, please instead search on the string "Black Magic Trace Capture" (`i->iInterface` to grab the appropriate string descriptor index) and validate there is a single IN endpoint. This will be the trace interface and will additionally allow Orbuculum to handle when the firmware it's looking at has had the trace interface omitted.

**Review by dragonmux — CHANGES_REQUESTED** [review]
This definitely looks better than presuming that trace is always a fixed endpoint. However, the trace interface may also be omitted and may (though this is somewhat more unlikely) move interface numbers in the future.

Going for a descriptor dive approach as you have here is definitely the right direction, it just needs applying more to get this to be robust.

File: Src/orbtraceIf.c
Comment (robots) [reply Src/orbtraceIf.c:489]: Good point! Didnt think about ommited trace functionality. I will rework it in the evening.

**Comment by ALTracer** [comment]
Chiming in as the secondary author of upstream change in PR1827 (the idea was not new): simply swapping 0x85 to 0x83 will not fly, there is a substantial amount of genuine hardware with original firmware which forks fine, and users shouldn't need to upgrade their probe firmware to continue using new orbuculum suite tools. But e.g. on F4 OTG_FS there will never be a working EP 0x85, and I tried doing something about it.
Please walk the descriptors via libusb functions and match on some/all of: Interface 5, Endpoint single IN any number, name "Black Magic Trace Capture", and generally other info in "common/usb_descriptors.h". Prefer responses/solutions by dragonmux, of course.

File: Src/orbtraceIf.c
Comment (robots) [reply Src/orbtraceIf.c]: done

**Comment by robots** [comment]
@ALTracer I have done the change as @dragonmux suggested. It searches for interface with "Black Magic Trace Capture" name.

File: Src/orbtraceIf.c
Comment (dragonmux) [inline Src/orbtraceIf.c:485]: Do note that `config->bNumInterfaces` is a uint8_t, and so this does signed-unsigned conversion which is best avoided.

File: Src/orbtraceIf.c
Comment (dragonmux) [inline Src/orbtraceIf.c:480]: Might be worth while using `libusb_error_name()` to present err to the user  as both a number and in a slightly more friendly way.

**Review by dragonmux — APPROVED** [review]
We've a couple of small notes about error reporting and implicit conversions, but otherwise this approach LGTM now 👍🏼

File: Src/orbtraceIf.c
Comment (robots) [reply Src/orbtraceIf.c:480]: dtto, see lines 530.

File: Src/orbtraceIf.c
Comment (dragonmux) [reply Src/orbtraceIf.c:485]: Aye, they're review items for things we'd do differently but we are not the Orbuculum maintainers (we are only the lead maintainer for BMD, so our review was purely about making things robust against BMD) which is why we approved with comments - it's up to Mubes and zyp if they want these items changed/addressed but we wouldn't have felt right not pointing them out to be able to have a conversation about.

Comment (robots) [reply Src/orbtraceIf.c:485]: I guess for this PR it should be ok, and there should be another PR. I will make new one, once this is merged and time permits. Executing cppcheck on the orbuculum codebase revelals more problems.

**Pull request merged**
